app/controllers/brexit_feedback_controller.rb

```diff
@@ -0,0 +1,58 @@
+require 'date'
+
+require 'google/apis/drive_v2'
+require 'google/api_client/client_secrets'
+
+require 'signet/oauth_2/client'
+
+class BrexitFeedbackController < ApplicationController
+  def home
+    session.destroy
+  end
+
+  def results
+    client_hash = session['auth_client']
+
+    opts = {
```

Comment:
I think some of this could probably be memoised but as it stands the use case for this is quite small. Essentially it's probably only going to be run once a week before being edited manually so I've just done this for clarity.

Reply:
The usual goal with Rails controller code is to keep as small a code footprint as possible.
Calling a service class or PORO to create this hash would be preferable.
It's also easier to unit test.

Reply:
okay, thanks, I'll clean this up some

Reply:
👍